### PR TITLE
Silence graphql_definition deprecation warning before suporting GQL 2.0

### PR DESCRIPTION
**Pull request by janraasch**
**goal** Silence `graphql_definition` deprecation warning for a v0.18.x release. 

**reasoning** Nothing users of this gem can do about it before a v1.0 release that would include support for GraphQL 2.0.

**context** 

* This follows the instructions by @mcelicalderon on https://github.com/graphql-devise/graphql_devise/issues/212#issuecomment-1061956457.
* This is based on https://github.com/graphql-devise/graphql_devise/pull/215

### CODE REVIEW

**Comment by janraasch** [comment]
Hi @mcelicalderon, I gave it a go over here following your instructions and cherry-picking what I could use from `master`.

**⚠️ beware** I have no idea why these two ruby 2.5 tests fail for GQL 1.13, see e.g. https://app.circleci.com/pipelines/github/graphql-devise/graphql_devise/148/workflows/87788dbb-b9ac-4cfd-891a-3ed6bde3feed/jobs/9211. The code is here: https://github.com/rmosolgo/graphql-ruby/blob/1.13.x/lib%2Fgraphql%2Fschema%2Ffield.rb#L282.

ℹ️ I saw that you skipped those tests on `master`, so I cherry-picked that commit 😇.

File: lib/graphql_devise/schema_plugin.rb [hunk 1]
```diff
@@ -113,7 +113,11 @@ def authenticate_option(field, trace_data)
       auth_required = if trace_data[:context]
         field.metadata[:authenticate]
       else
-        field.graphql_definition.metadata[:authenticate]
+        if Gem::Version.new(GraphQL::VERSION) >= Gem::Version.new('1.13.1')
```
Comment (mcelicalderon) [inline lib/graphql_devise/schema_plugin.rb:116]: ✅

File: Appraisals [hunk 2]
```diff
@@ -82,20 +82,6 @@ appraise 'rails5.2-graphql1.12' do
   gem 'rspec-rails', '< 4.0'
 end
 
-appraise 'rails6.0-graphql1.8' do
```
Comment (mcelicalderon) [inline Appraisals:85]: 🙌

**Review by mcelicalderon — COMMENTED** [review]
Thank you, @janraasch, this looks great!

> beware I have no idea why these two ruby 2.5 tests fail for GQL 1.13

I was able to replicate that locally, but still couldn't get to the bottom of this, happens when the graphql gem is required, so might just be test related. Anyway, not a concer of this MR.

**Review by mcelicalderon — APPROVED** [review]
Thanks, I'll push a new release with this changes asap

**Comment by mcelicalderon** [comment]
[v0.18.2](https://rubygems.org/gems/graphql_devise/versions/0.18.2) released! Thank you for the contribution, @janraasch!

**Pull request merged**